examples/transformers_text.py

```diff
@@ -194,6 +245,22 @@ def mean_pooling(last_hidden_state: Tensor, attention_mask) -> Tensor:
     return embedding.unsqueeze(1)
 
 
+def last_pooling(last_hidden_state: Tensor, attention_mask: Tensor) -> Tensor:
+    left_padding = (attention_mask[:, -1].sum() == attention_mask.shape[0])
+    if left_padding:
+        return last_hidden_state[:, -1]
+    else:
```

Comment:
and I feel `left_padding` is rarely true in most cases any way.

Reply:
please ignore this comment if this `left_padding` thing is a standard thing to do in LLM community. I am just wondering.

Reply:
Right for decoder architecture model (causal LLM) I think most cases we will use left padding (each token attends only attends to previous tokens, so that's why we use the last token embedding as the sentence embedding)

Reply:
Thanks. sorry, I still don't fully understand. even if `left_padding` is False, we still use the past token embedding, right?

Reply:
Maybe I am missing something. Can you please add some shape annotation and comments?

Reply:
Oh, I see what you mean. Right, `left_padding` just makes a little bit faster. Just using the `else` one should be ok. Removed the one in the `if`